python/ray/data/tests/test_dynamic_block_split.py

```diff
@@ -3,67 +3,268 @@
 import pytest
 
 import ray
+from ray.data._internal.lazy_block_list import LazyBlockList
 from ray.data.block import BlockMetadata
 from ray.data.datasource import Datasource
 from ray.data.datasource.datasource import ReadTask, Reader
 
 from ray.tests.conftest import *  # noqa
 
 
-def test_read_large_data(ray_start_cluster):
-    # Test 20G input with single task
-    num_batch = 20
+# Data source generates random bytes data
+class RandomBytesDatasource(Datasource):
+    def create_reader(self, **read_args):
+        return RandomBytesReader(read_args["num_blocks"], read_args["block_size"])
+
+
+class RandomBytesReader(Reader):
+    def __init__(self, num_blocks: int, block_size: int):
+        self.num_blocks = num_blocks
+        self.block_size = block_size
+
+    def estimate_inmemory_data_size(self):
+        return None
+
+    def get_read_tasks(self, parallelism: int):
+        def _blocks_generator():
+            for _ in range(self.num_blocks):
+                yield pd.DataFrame({"one": [np.random.bytes(self.block_size)]})
+
+        return parallelism * [
+            ReadTask(
+                lambda: _blocks_generator(),
+                BlockMetadata(
+                    num_rows=self.num_blocks,
+                    size_bytes=self.num_blocks * self.block_size,
+                    schema=None,
+                    input_files=None,
+                    exec_stats=None,
+                ),
+            )
+        ]
+
+
+def test_dataset(ray_start_regular_shared, enable_dynamic_block_splitting):
+    # Test 10 blocks from 10 tasks, each block is 1024 bytes.
+    num_blocks = 10
+    block_size = 1024
+    num_tasks = 10
+
     ctx = ray.data.context.DatasetContext.get_current()
-    block_splitting_enabled = ctx.block_splitting_enabled
-    ctx.block_splitting_enabled = True
+    target_max_block_size = ctx.target_max_block_size
+    ctx.target_max_block_size = block_size
```

Comment:
Instead of manually manipulating the context and having this big try-except block, could we use a fixture for this? E.g. https://github.com/ray-project/ray/blob/64cb5bb7b735b57c881e43b26680f98b90978a87/python/ray/data/tests/conftest.py#L281-L287

This could be something like
```python
@pytest.fixture(params=[1024])
def target_max_block_size(request):
    ctx = ray.data.context.DatasetContext.get_current()
    original = ctx.target_max_block_size
    ctx.target_max_block_size = request.param
    yield request.param
    ctx.target_max_block_size = original
```

Reply:
@clarkzinzow - sure, changed.